geoh5py/io/h5_reader.py

```diff
@@ -291,6 +309,16 @@ def fetch_values(cls, h5file: Optional[str], uid: uuid.UUID) -> Optional[float]:
         name = list(project.keys())[0]
         if "Data" in list(project[name]["Data"][cls.uuid_str(uid)].keys()):
             values = np.r_[project[name]["Data"][cls.uuid_str(uid)]["Data"]]
+            if not isinstance(values[0], (str, bytes)):
+                if values.dtype in [float, "float64", "float32"]:
+                    ind = values == FLOAT_NDV
+                else:
+                    ind = values == INTEGER_NDV
+                    values = values.astype("float64")
```

Comment:
**question:** can really a float64 hold any integer values? Should it not store them in an array of integers ?

It is unlikely the data uses the full min,max range of integer values, but in theory it could use some integer values that do not fit in a float64 representation

Reply:
Issue is how to handle the NDV. Unfortunately, numpy doesn't support nan on integer type. My thought was to convert everything to float, then transfer back on the write?

Reply:
not sure to follow: NDV for integer data is an int (-2147483648)

Reply:
I meant on the python side ... it is nicer to work with numpy.nan, but that's not supported if the array is type int32...

Reply:
ok, seems fine then. float64 seems able to store any value from int32